### ISSUE TITLE
settings.xml file format changed

### ISSUE THREAD

**Issue by Almighty188** [winsiderss/systeminformer#2202]
### Brief description of your issue

I am using systeminformer-3.1.24258-canary-bin with the SystemInformer.exe.settings.xml file from systeminformer-3.1.24244-canary-bin which worked on ysteminformer-3.1.24244-canary-bin without issues but with systeminformer-3.1.24258-canary-bin, it appears to be ignoring everything in the file.  Did the format or options in the file change?  

### Steps to reproduce (optional)

Use a existing  SystemInformer.exe.settings.xml file

### Expected behavior (optional)

It is supposed to use the settings in the file.

### Actual behavior (optional)

All settings are ignored and System Informer seems to use the default settings.  In fact, it seems even if one uses the default  SystemInformer.exe.settings.xml file and then configure things, it seems to be reading and saving the settings elsewhere and not to  SystemInformer.exe.settings.xml 

### Environment (optional)

```shell
systeminformer-3.1.24258-canary-bin
Windows 11 x64 Professional Edition 23H2 Build 22635.4145 Beta Insiders
```

**Comment by dmex** [comment]
> Did the format or options in the file change?

Nothing changed.

> Use a existing SystemInformer.exe.settings.xml file

I just tested this and no issues?

**Comment by Almighty188** [comment]
Something changed because let me show you something I didn't mention, as I have closed System Informer and then opened it again and this is what my settings looks like on screen which means I obviously have different things enabled on screen as you can tell form the columns itself and what's on the bottom status bar for example.

![image](https://github.com/user-attachments/assets/1a2deddf-3da5-44c3-889b-28b81a491add)

Now, please take a look at the contents of the SystemInformer.exe.settings.xml file itself.  Do you even see anything other than the defaults in it?  Where is it storing the other settings since it seems to be reading the configuration from somewhere else.

![image](https://github.com/user-attachments/assets/06eae547-f529-408e-959a-c14902388522)

It is reading and savings somewhere else other than C:\Users\vince\Downloads\systeminformer-3.1.24258-canary-bin\amd64\SystemInformer.exe.settings.xml

**Comment by dmex** [comment]
> Where is it storing the other settings since it seems to be reading the configuration from somewhere else.

The same default location we've been using for years?

```
%APPDATA%\SystemInformer\settings.xml
```

> Also, where is the GeoIP ID and license key actually stored specifically?

The settings file?

**Comment by Almighty188** [comment]
> > Where is it storing the other settings since it seems to be reading the configuration from somewhere else.
> 
> The same default location we've been using for years?
> 
> ```
> %APPDATA%\SystemInformer\settings.xml
> ```
> 
> > Also, where is the GeoIP ID and license key actually stored specifically?
> 
> The settings file?

I thought if one ran it as a portable which is extracting the zip file to a directory of my choice, it would be using the systeminformer-*version*-bin\amd64\SystemInformer.exe.settings.xml file and then running SystemInformer.exe as I have always replaced only that file and I would have the exact settings without manually configuring System Informer and if I didn't, it would just be the default because there has to be a reason that the amd64 directory contains a SystemInformer.exe.settings.xml file by default.  

and this confirms what I just wrote:
https://github.com/winsiderss/systeminformer

![image](https://github.com/user-attachments/assets/eaafb57d-4bce-467e-b86a-96c90c6c8fc8)

So basically the issue is atleast in my case, it is not reading or writing to the SystemInformer.exe.settings.xml in the same directory where the SystemInformer.exe is.  Remember I didn't write the screenshot above....

**Comment by Almighty188** [comment]
So I have confirmed, it is not reading the SystemInformer.exe.settings.xml file from the same directory as SystemInformer.exe like it did in previous builds which is actually in the instructions as in lines 51-62 in README.md as previously mentioned.

**Comment by MagicAndre1981** [comment]
For me the xml is still used on all systems.

run ProcMon and trace SI startup and look at IO events when settings are read if you see any errors in Result column.

**Comment by Almighty188** [comment]
> For me the xml is still used on all systems.
> 
> run ProcMon and trace SI startup and look at IO events when settings are read if you see any errors in Result column.

Thanks for the pointers, it only happens with this build as I reverted to the previous build with the same SystemInformer.exe.settings.xml file in program directory and it works fine and that's with  %APPDATA%\SystemInformer deleted.
To be honest, I still don't know how to use functions in System Informer and obviously not ProcMon or Process Explorer for that matter but how exactly do I do what you mentioned as I am running ProcMon from wscc.

**Comment by dmex** [comment]
> As I do not know if https://github.com/winsiderss/systeminformer/issues/1 works or not as I never tried it. We know https://github.com/winsiderss/systeminformer/pull/3 works. https://github.com/winsiderss/systeminformer/issues/2 is what I am referring to and one of these commits dated September 8, 2024 and later to main.c excluding July 3, 2024 probably is responsible for the issue I mentioned,

None of these commits changed the settings.

> So can you give step by step instructions as always good to learn something new...

Search for the settings.xml filename in procmon and right-click the filename (must be the filename) then select Include - this will exlude everything but the file and there should be 15 or so results on screen. Make sure the full string for the filename is visible and the status column.

**Comment by MagicAndre1981** [comment]
In the Filter select `SystemInformer.exe` as process to Include and only show disk and thread/process events (registry, Network, Profiling can be disabled from toolbar)

![image](https://github.com/user-attachments/assets/9b360875-7392-4dd6-a931-2a2b0ee2f6cf)

Click to ok to apply filter, clear view and run the new systeminformer.exe build. In ProcMon you can configure symbols, add here the Si symbols and now look in result column for non success events.

**Comment by Almighty188** [comment]
@MagicAndre1981 - A few questions... 

Basically what I did was select Filter and then SystemInformer.exe as you mentioned.  Then on the toolbar at the top, I have only the 2nd and 4th icons selected, am I supposed to let it do the Applying Event Filter process when I click on the 2nd and 4th icon.  
Am I supposed to do the toolbar at the top after the Filter part or before.  And am I correct the symbols file for SI is in systeminformer-3.1.24258-pdb.zip and then do I need to click the Capture button on the top before running systeminformer.exe.  I also renamed the %APPDATA%\SystemInformer\settings.xml to settings.xml.1 first just to make sure it doesn't use it.

**Comment by MagicAndre1981** [comment]
in the symbols entry and `;C:\sym\systeminformer-3.1.24258-pdb\` to add the localy expanded PDBs

**Comment by Almighty188** [comment]
Not sure what I am doing wrong but basically the Configure symbols screen is as above, I have the filter as this is what it shows when I open the filter again after clicking OK.

![image](https://github.com/user-attachments/assets/c1e0f1e8-b0b8-4af9-9c94-f0cd570e5681)

I basically clicked on the Capture button and then ran SystemInformer.exe but the procmon page is blank.

![image](https://github.com/user-attachments/assets/b6cf572f-2b0d-47de-9414-5cbbfbd28b54)

**Comment by Almighty188** [comment]
Never mind, I have it on Architecture instead of Process Name.

**Comment by Almighty188** [comment]
This is what it shows, is it supposed to be 33,641 events? 

![image](https://github.com/user-attachments/assets/44a5116a-3de1-4ea5-ab85-11db0e487749)

**Comment by Almighty188** [comment]
Apparently, now it works fine

![image](https://github.com/user-attachments/assets/e3b8764a-db3b-4491-9d9f-c6ab65810d4f)

and what's interesting is unlike before, the %APPDATA%\SystemInformer directory no longer creates a settings.xml on it's own like it did before when I ran SystemInformer.exe after settings.xml was already renamed and only the settings.xml1 file which is what I renamed the settings.xml to before running SystemInformer.exe

**Comment by dmex** [comment]
Ignore the symbols they're not relevant. You need to filer by the settings.xml filename because that's the only file you're having issues with. If the window is blank then you're not using the correct filename.

You're trying to filter by process architecture with a filername?

![image](https://github.com/user-attachments/assets/06ff1b1a-2ca8-433d-bc3b-5b6f86286c55)

Change to `Path` and `Ends With` file `settings.xml`

![image](https://github.com/user-attachments/assets/dc0ae458-569e-48bb-980f-e263d2e78b57)

**Comment by Almighty188** [comment]
@dmex, actually as I mentioned in two posts above.  As soon as I got procmon working, something changed as I said:
"%APPDATA%\SystemInformer directory no longer creates a settings.xml on it's own like it did before when I ran SystemInformer.exe after settings.xml was already renamed as settings.xml1 and only the settings.xml1 file exists in that directory as there is no cache or the GeoLite2-City.mmdb yet which is what I renamed the settings.xml to before running SystemInformer.exe

I did not reboot other than opened and closed SystemInformer.exe a few times that was started with Run as Adminstrator because I had the filter on Architecture instead of Process Name for procmon.

And after doing the GeoIP database update, the cache folder is created in the same directory as SystemInformer.exe and no longer in %APPDATA%\SystemInformer while the GeoLite2-City.mmdb is in %APPDATA%\SystemInformer so it's a mystery what actually happened."

so something happened as the cache folder is now created in the SystemInformer.exe folder instead of %APPDATA%\SystemInformer and it reads SystemInformer.exe.settings.xml file from the SystemInformer.exe directory correctly and it no longer creates the settings.xml file in %APPDATA%\SystemInformer.
